README.rst

```diff
@@ -167,12 +168,85 @@ Additionally Supported Parameters
 |                 |                                                                             |
 |                 | Defaults to the truncated description.                                      |
 +-----------------+-----------------------------------------------------------------------------+
-|       url       | Adds a URL in email as <br/>                                                | 
+|       url       | Adds a URL in email as <br/>                                                |
 |                 |                                                                             |
 |                 | ``For more information see <url>.`` <br/>                                   |
-|                 |                                                                             | 
+|                 |                                                                             |
 |                 | Default to **None** meaning above message would not be added to the email.  |
 +-----------------+-----------------------------------------------------------------------------+
+|       type      | Set values of other parameters based on predefined setting                  |
+|                 | ``OPENWISP_NOTIFICATION_TYPES``                                             |
+|                 |                                                                             |
+|                 | Defaults to **None** meaning you need to provide other arguments.           |
++-----------------+-----------------------------------------------------------------------------+
+
+Settings
+--------
+
+``OPENSWISP_NOTIFICATION_TYPES``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    
+
++--------------+-------------+
+| **type**:    | ``dict``    |
++--------------+-------------+
+| **default**: | ``{}``      |
++--------------+-------------+
+
+This setting allows to define additional notification types.
+
+For example, if you want to add a notification type ``device added`` you can use:
+
+.. code-block:: python
+
+    # In your_project/settings.py
+    OPENSWISP_NOTIFICATION_TYPES = {
+        'custom type': {
+            'level': 'info',
+            'verb': 'added',
+            'name': 'device added',
+            'message': 'configurables/message.md',
+        },
+    }
+
+``OPENWISP_NOTIFICATION_MESSAGE_TEMPLATE``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+--------------------------------+
+| **type**:    | ``string``                     |
++--------------+--------------------------------+
+| **default**: | ``configurables/message.md``   |
++--------------+--------------------------------+
+
+This setting allows to provide a default markdown formatted message template.
+If a notification type does not define it's message template explicitly, then this message template will be used.
```

Comment:
I think a notification type must define its message and subject unless we find a use case that forces us to give up this requirement. At the moment I don't see this use case, if you see one, let me know.

Reply:
The content of [configurables/message.md](https://github.com/openwisp/openwisp-notifications/pull/19/files#diff-50457a64c07afe611379ffa9e1a20212) is very generic as of now. It can work for every notification. This serves as a fallback setting if `message` is not defined  for a particular notification type. 

Use Case:
For monitoring module which currently implements only two notification types can use a common template for both. This will help us from repeating ourselves while defining a notification type.